src/SIM/SIMoutput.C

```diff
@@ -587,10 +587,15 @@ int SIMoutput::writeGlvS1 (const Vector& psol, int iStep, int& nBlock,
           field.fillColumn(j,pSol(Vec4(*cit,time)).ptr());
         if (mySol->getScalarSol())
         {
-          cit = grid->begin_XYZ();
-          const RealFunc& sSol = *mySol->getScalarSol();
-          for (j = 1; cit != grid->end_XYZ() && haveXsol; j++, ++cit)
-            field(field.rows(),j) = sSol(Vec4(*cit,time,grid->getParam(j-1)));
+          size_t r = pSol.dim();
+          const RealFunc* psSol = mySol->getScalarSol(0);
+          for (size_t k = 0; psSol; k++) {
```

Comment:
Suggest following (cosmetic) rewrite, including the subsequent two comments:
```
  const RealFunc* psSol;
  size_t r = pSol.dim()+1;
  for (size_t k = 1; (psSol = mySol->getScalarSol(k)); k++, r++)
  {
```

Reply:
changed to this, with a fix (need to be k-1 in the getScalarSol call).

Reply:
Right, but then maybe better to let the loop start with k = 0 and you save one operation (since k is not used elsewhere)